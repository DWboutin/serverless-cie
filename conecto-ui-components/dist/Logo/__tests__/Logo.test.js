"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _react = _interopRequireDefault(require("react"));

var _enzyme = require("enzyme");

var _Logo = _interopRequireDefault(require("../../Logo"));

var _Conecto = _interopRequireDefault(require("../Logos/Conecto"));

var _To = _interopRequireDefault(require("../Logos/To"));

describe('Logo', () => {
  describe('SVG components', () => {
    it('Conecto component should match snapshot', () => {
      const wrapper = (0, _enzyme.shallow)(_react.default.createElement(_Conecto.default, null));
      expect(wrapper).toMatchSnapshot();
    });
    it('Short Conecto component should match snapshot', () => {
      const wrapper = (0, _enzyme.shallow)(_react.default.createElement(_To.default, null));
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('Default', () => {
    it('should match snapshot', () => {
      const wrapper = (0, _enzyme.shallow)(_react.default.createElement(_Logo.default, null));
      expect(wrapper).toMatchSnapshot();
    });
    it('should have the prefix class', () => {
      const wrapper = (0, _enzyme.shallow)(_react.default.createElement(_Logo.default, null));
      expect(wrapper.instance().classesPrefix).toEqual('to-logo');
      expect(wrapper.hasClass(wrapper.instance().classesPrefix)).toEqual(true);
    });
    it('should contain added classes', () => {
      const wrapper = (0, _enzyme.shallow)(_react.default.createElement(_Logo.default, {
        className: "test-class-1 test-class-2"
      }));
      expect(wrapper.hasClass('test-class-1')).toEqual(true);
      expect(wrapper.hasClass('test-class-2')).toEqual(true);
    });
    it('should render the SVG component of Conecto', () => {
      const wrapper = (0, _enzyme.mount)(_react.default.createElement(_Logo.default, null));
      expect(wrapper.find(_Conecto.default).length).toEqual(1);
    });
  });
  describe('Short', () => {
    it('should match snapshot', () => {
      const wrapper = (0, _enzyme.shallow)(_react.default.createElement(_Logo.default, {
        size: "short"
      }));
      expect(wrapper).toMatchSnapshot();
    });
    it('should have the prefix classes', () => {
      const wrapper = (0, _enzyme.shallow)(_react.default.createElement(_Logo.default, {
        size: "short"
      }));
      expect(wrapper.instance().classesPrefix).toEqual('to-logo');
      expect(wrapper.hasClass(wrapper.instance().classesPrefix)).toEqual(true);
      expect(wrapper.hasClass('to-logo-short')).toEqual(true);
    });
    it('should render the SVG component of the short version of Conecto', () => {
      const wrapper = (0, _enzyme.mount)(_react.default.createElement(_Logo.default, {
        size: "short"
      }));
      expect(wrapper.find(_To.default).length).toEqual(1);
      wrapper.unmount();
    });
  });
  describe('With Suffix', () => {
    it('should match snapshot', () => {
      const wrapper = (0, _enzyme.shallow)(_react.default.createElement(_Logo.default, {
        suffix: "Contractor"
      }));
      expect(wrapper).toMatchSnapshot();
    });
    it('should have the prefix classes', () => {
      const wrapper = (0, _enzyme.shallow)(_react.default.createElement(_Logo.default, {
        suffix: "Contractor"
      }));
      expect(wrapper.instance().classesPrefix).toEqual('to-logo');
      expect(wrapper.hasClass(wrapper.instance().classesPrefix)).toEqual(true);
      expect(wrapper.hasClass('to-logo-contractor')).toEqual(true);
      expect(wrapper.hasClass('to-logo-with-suffix')).toEqual(true);
    });
    it('should render correctly the text suffix', () => {
      const wrapper = (0, _enzyme.mount)(_react.default.createElement(_Logo.default, {
        suffix: "Contractor"
      }));
      expect(wrapper.text()).toEqual('Contractor');
      wrapper.unmount();
    });
    it('should render correctly a tag suffix', () => {
      const wrapper = (0, _enzyme.mount)(_react.default.createElement(_Logo.default, {
        suffix: _react.default.createElement("strong", null, "ABcdEF123")
      }));
      expect(wrapper.find('strong').length).toEqual(1);
      expect(wrapper.text()).toEqual('ABcdEF123');
      wrapper.unmount();
    });
  });
  describe('With Type', () => {
    it('should match snapshot', () => {
      const wrapper = (0, _enzyme.shallow)(_react.default.createElement(_Logo.default, {
        type: "roofing",
        suffix: "Roofing"
      }));
      expect(wrapper).toMatchSnapshot();
    });
    it('should have the prefix classes', () => {
      const wrapper = (0, _enzyme.shallow)(_react.default.createElement(_Logo.default, {
        type: "roofing",
        suffix: "Roofing"
      }));
      expect(wrapper.instance().classesPrefix).toEqual('to-logo');
      expect(wrapper.hasClass(wrapper.instance().classesPrefix)).toEqual(true);
      expect(wrapper.hasClass('to-logo-roofing')).toEqual(true);
      expect(wrapper.hasClass('to-logo-with-suffix')).toEqual(true);
    });
    it('should render correctly the text suffix', () => {
      const wrapper = (0, _enzyme.mount)(_react.default.createElement(_Logo.default, {
        type: "roofing",
        suffix: "Roofing"
      }));
      expect(wrapper.text()).toEqual('Roofing');
      wrapper.unmount();
    });
  });
});