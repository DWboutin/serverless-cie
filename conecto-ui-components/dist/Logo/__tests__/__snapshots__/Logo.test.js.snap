// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo Default should match snapshot 1`] = `
<div
  className="to-logo"
>
  <span
    className="to-logo-wrapper"
  >
    <Conecto />
  </span>
</div>
`;

exports[`Logo SVG components Conecto component should match snapshot 1`] = `
<svg
  className="to-logo-conecto"
  height="20px"
  viewBox="0 0 115 20"
  width="115px"
>
  <path
    d="M7.696875,15.2275 C6.003125,15.2275 4.930625,14.0425 4.930625,12.095 C4.930625,10.203125 6.059375,8.9325 7.696875,8.9325 C8.9675,8.9325 10.04,9.581875 10.2375,10.936875 L15.008125,10.936875 C14.64125,7.239375 11.79,4.755 7.66875,4.755 C3.434375,4.755 0.30125,7.775 0.30125,12.095 C0.30125,16.41375 3.434375,19.434375 7.66875,19.434375 C11.903125,19.434375 14.83875,16.80875 15.03625,12.94125 L10.29375,12.94125 C10.096875,14.49375 9.02375,15.2275 7.696875,15.2275"
    fill="#572D2D"
  />
  <path
    d="M24.198125,15.25625 C22.531875,15.25625 21.31875,13.985625 21.31875,12.095 C21.31875,10.203125 22.531875,8.9325 24.198125,8.9325 C25.89125,8.9325 27.105,10.203125 27.105,12.095 C27.105,13.985625 25.89125,15.25625 24.198125,15.25625 Z M24.198125,4.755 C19.906875,4.755 16.716875,7.719375 16.716875,12.095 C16.716875,16.47 19.906875,19.434375 24.198125,19.434375 C28.48875,19.434375 31.678125,16.47 31.678125,12.095 C31.678125,7.719375 28.48875,4.755 24.198125,4.755 L24.198125,4.755 Z"
    fill="#572D2D"
  />
  <path
    d="M42.333125,4.755 C40.52625,4.755 39.029375,5.57375 38.183125,7.041875 L38.183125,5.0375 L33.525,5.0375 L33.525,19.151875 L38.324375,19.151875 L38.324375,11.304375 C38.324375,9.46875 39.08625,8.565625 40.44125,8.565625 C41.824375,8.565625 42.558125,9.46875 42.558125,11.304375 L42.558125,19.151875 L47.3575,19.151875 L47.3575,10.14625 C47.3575,6.533125 45.211875,4.755 42.333125,4.755"
    fill="#572D2D"
  />
  <path
    d="M53.608125,10.82375 C53.91875,9.27125 54.87875,8.480625 56.34625,8.480625 C57.814375,8.480625 58.8025,9.215 59.113125,10.82375 L53.608125,10.82375 Z M56.34625,4.755 C52.1125,4.755 49.06375,7.775 49.06375,12.095 C49.06375,16.41375 52.1125,19.434375 56.34625,19.434375 C59.96,19.434375 62.415625,17.740625 63.319375,14.49375 L58.7175,14.49375 C58.350625,15.340625 57.560625,15.708125 56.459375,15.708125 C54.8225,15.708125 53.8625,14.88875 53.58,13.223125 L63.573125,13.223125 C63.60125,12.85625 63.63,12.489375 63.63,12.095 C63.63,7.775 60.60875,4.755 56.34625,4.755 L56.34625,4.755 Z"
    fill="#572D2D"
  />
  <path
    d="M72.70625,15.2275 C71.0125,15.2275 69.94,14.0425 69.94,12.095 C69.94,10.203125 71.06875,8.9325 72.70625,8.9325 C73.976875,8.9325 75.049375,9.581875 75.246875,10.936875 L80.0175,10.936875 C79.650625,7.239375 76.799375,4.755 72.678125,4.755 C68.44375,4.755 65.31,7.775 65.31,12.095 C65.31,16.41375 68.44375,19.434375 72.678125,19.434375 C76.9125,19.434375 79.848125,16.80875 80.045625,12.94125 L75.303125,12.94125 C75.105625,14.49375 74.033125,15.2275 72.70625,15.2275"
    fill="#572D2D"
  />
  <path
    d="M103.67625,14.309375 C103.15875,14.909375 102.4075,15.2575 101.52125,15.2575 C100.134375,15.2575 99.065,14.379375 98.741875,12.995 C98.676875,12.715 98.641875,12.413125 98.641875,12.095625 L98.641875,12.03875 C98.66625,10.180625 99.870625,8.93375 101.52125,8.93375 C102.4075,8.93375 103.15875,9.28125 103.67625,9.881875 C104.1525,10.42875 104.4275,11.193125 104.4275,12.095625 C104.4275,12.9975 104.1525,13.7625 103.67625,14.309375 Z M103.67625,5.025625 C103,4.848125 102.278125,4.755625 101.52125,4.755625 C97.249375,4.755625 94.06625,7.691875 94.039375,12.03875 L94.039375,12.095625 C94.006875,13.5175 92.840625,14.6675 91.410625,14.6675 C89.961875,14.6675 88.784375,13.4875 88.784375,12.03875 L88.784375,10.934375 L91.94625,10.934375 C92.145625,9.11125 92.805625,7.4875 93.829375,6.17 L88.784375,6.17 L88.784375,0.565 L84.0175,0.565 L84.0175,6.17 L81.170625,6.17 L81.170625,10.934375 L84.0175,10.934375 L84.0175,12.03875 C84.0175,16.11625 87.333125,19.435 91.410625,19.435 C93.29875,19.435 95.0225,18.72375 96.33125,17.551875 C97.650625,18.750625 99.458125,19.435 101.52125,19.435 C102.278125,19.435 103,19.343125 103.67625,19.165 C106.838125,18.346875 109.000625,15.701875 109.000625,12.095625 C109.000625,8.48875 106.838125,5.844375 103.67625,5.025625 L103.67625,5.025625 Z"
    fill="#572D2D"
  />
  <path
    d="M112.21625,14.49375 C110.7775,14.49375 109.7325,15.565625 109.7325,16.95 C109.7325,18.331875 110.7775,19.43375 112.21625,19.43375 C113.656875,19.43375 114.699375,18.331875 114.699375,16.95 C114.699375,15.565625 113.656875,14.49375 112.21625,14.49375"
    fill="#572D2D"
  />
</svg>
`;

exports[`Logo SVG components Short Conecto component should match snapshot 1`] = `
<svg
  className="to-logo-to"
  height="20px"
  viewBox="0 0 34 20"
  width="34px"
>
  <path
    d="M22.505625,13.744375 C21.988125,14.344375 21.236875,14.6925 20.350625,14.6925 C18.96375,14.6925 17.894375,13.814375 17.57125,12.43 C17.50625,12.15 17.47125,11.848125 17.47125,11.530625 L17.47125,11.47375 C17.495625,9.615625 18.7,8.36875 20.350625,8.36875 C21.236875,8.36875 21.988125,8.71625 22.505625,9.316875 C22.981875,9.86375 23.256875,10.628125 23.256875,11.530625 C23.256875,12.4325 22.981875,13.1975 22.505625,13.744375 Z M22.505625,4.460625 C21.829375,4.283125 21.1075,4.190625 20.350625,4.190625 C16.07875,4.190625 12.895625,7.126875 12.86875,11.47375 L12.86875,11.530625 C12.83625,12.9525 11.67,14.1025 10.24,14.1025 C8.79125,14.1025 7.61375,12.9225 7.61375,11.47375 L7.61375,10.369375 L10.775625,10.369375 C10.975,8.54625 11.635,6.9225 12.65875,5.605 L7.61375,5.605 L7.61375,0 L2.846875,0 L2.846875,5.605 L5.32907052e-14,5.605 L5.32907052e-14,10.369375 L2.846875,10.369375 L2.846875,11.47375 C2.846875,15.55125 6.1625,18.87 10.24,18.87 C12.128125,18.87 13.851875,18.15875 15.160625,16.986875 C16.48,18.185625 18.2875,18.87 20.350625,18.87 C21.1075,18.87 21.829375,18.778125 22.505625,18.6 C25.6675,17.781875 27.83,15.136875 27.83,11.530625 C27.83,7.92375 25.6675,5.279375 22.505625,4.460625 L22.505625,4.460625 Z"
    fill="#572D2D"
  />
  <path
    d="M31.045625,13.92875 C29.606875,13.92875 28.561875,15.000625 28.561875,16.385 C28.561875,17.766875 29.606875,18.86875 31.045625,18.86875 C32.48625,18.86875 33.52875,17.766875 33.52875,16.385 C33.52875,15.000625 32.48625,13.92875 31.045625,13.92875"
    fill="#572D2D"
  />
</svg>
`;

exports[`Logo Short should match snapshot 1`] = `
<div
  className="to-logo to-logo-short"
>
  <span
    className="to-logo-wrapper"
  >
    <To />
  </span>
</div>
`;

exports[`Logo With Suffix should match snapshot 1`] = `
<div
  className="to-logo to-logo-contractor to-logo-with-suffix"
>
  <span
    className="to-logo-wrapper"
  >
    <Conecto />
  </span>
  <div
    className="to-logo-suffix"
  >
    Contractor
  </div>
</div>
`;

exports[`Logo With Type should match snapshot 1`] = `
<div
  className="to-logo to-logo-roofing to-logo-with-suffix"
>
  <span
    className="to-logo-wrapper"
  >
    <Conecto />
  </span>
  <div
    className="to-logo-suffix"
  >
    Roofing
  </div>
</div>
`;
